### PR TITLE
docs: adds CTE guide and references

### PR DESCRIPTION
**Pull request by killme2008**
## What's Changed in this PR

*Describe the change in this PR*

* Adds user guide and references for common table expression (CTE).

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/nightly/en/user-guide/query-data/cte.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+# Common Table Expression (CTE)
+
+CTEs are similar to [Views](./view.md) in that they help you reduce the complexity of your queries, break down long and complex SQL statements, and improve readability and reusability.
+
+You already read a CTE in the [quickstart](/getting-started/quick-start#correlate-metrics-and-logs) document.
+
+### What is a Common Table Expression (CTE)?
```
Comment (nicecui) [inline docs/nightly/en/user-guide/query-data/cte.md:7]:
```suggestion
## What is a Common Table Expression (CTE)?
```

File: docs/nightly/en/user-guide/query-data/cte.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+A Common Table Expression (CTE) is a temporary result set that you can reference within a `SELECT`, `INSERT`, `UPDATE`, or `DELETE` statement. CTEs help to break down complex queries into more readable parts and can be referenced multiple times within the same query.
+
+### Basic Syntax of CTE
```
Comment (nicecui) [inline docs/nightly/en/user-guide/query-data/cte.md:11]:
```suggestion
## Basic syntax of CTE
```

File: docs/nightly/en/user-guide/query-data/cte.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+```sql
+WITH cte_name [(column1, column2, ...)] AS (
+    QUERY
+)
+SELECT ...
+FROM cte_name;
+```
+
+### Example Explanation
```
Comment (nicecui) [inline docs/nightly/en/user-guide/query-data/cte.md:23]:
```suggestion
## Example explanation
```

File: docs/nightly/en/user-guide/query-data/cte.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+Next, we'll go through a complete example that demonstrates how to use CTEs, including data preparation, CTE creation, and usage.
+
+### Step 1: Create Example Data
```
Comment (nicecui) [inline docs/nightly/en/user-guide/query-data/cte.md:27]:
```suggestion
### Step 1: Create example data
```

File: docs/nightly/en/user-guide/query-data/cte.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    PRIMARY KEY(host, log_level),
+);
+
+INSERT INTO app_logs VALUES 
+('2023-10-01 10:00:00', 'host1', 'Error on service', 'ERROR'),
+('2023-10-01 10:00:00', 'host2', 'All services OK', 'INFO'),
+('2023-10-01 10:00:05', 'host1', 'Error connecting to DB', 'ERROR');
+```
+
+### Step 2: Define and Use CTEs
```
Comment (nicecui) [inline docs/nightly/en/user-guide/query-data/cte.md:61]:
```suggestion
### Step 2: Define and use CTEs
```

File: docs/nightly/en/user-guide/query-data/cte.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
++---------------------+-------+-------------+------------+
+| ts                  | host  | p95_latency | num_errors |
++---------------------+-------+-------------+------------+
+| 2023-10-01 10:00:00 | host2 |         150 |          0 |
+| 2023-10-01 10:00:00 | host1 |         120 |          1 |
+| 2023-10-01 10:00:05 | host1 |         130 |          1 |
++---------------------+-------+-------------+------------+
+```
+
+### Detailed Explanation
```
Comment (nicecui) [inline docs/nightly/en/user-guide/query-data/cte.md:111]:
```suggestion
### Detailed explanation
```

File: docs/nightly/en/user-guide/query-data/cte.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+            host, 
+            approx_percentile_cont(latency, 0.95) RANGE '5s' AS p95_latency
+          FROM 
+            grpc_latencies
+          ALIGN '5s' FILL PREV
+      ),
+      ```
+     Here we use a [range query](/user-guide/query-data/sql#aggregate-data-by-time-window) to calculate the 95th percentile latency for each `host` within every 5-second window.
+     
+    - `logs`:
```
Comment (nicecui) [inline docs/nightly/en/user-guide/query-data/cte.md:129]: The indentation here needs to be the same as `metrics`

File: docs/nightly/zh/user-guide/query-data/cte.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
+# 公共表表达式（CTE）
+
+CTE 与 [视图](./view.md) 类似，它们帮助您简化查询的复杂性，将长而复杂的 SQL 语句分解，并提高可读性和可重用性。
+
+您已经在 [快速开始](/getting-started/quick-start#指标和日志的关联查询) 文档中阅读了一个 CTE 的例子。
+
+### 什么是公共表表达式 （CTE）？
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/query-data/cte.md:7]:
```suggestion
## 什么是公共表表达式 （CTE）？
```

File: docs/nightly/zh/user-guide/query-data/cte.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+公共表表达式 (CTE) 是可以在 `SELECT`、`INSERT`、`UPDATE` 或 `DELETE` 语句中引用的临时结果集。CTE 有助于将复杂的查询分解成更可读的部分，并且可以在同一个查询中多次引用。
+
+### CTE 的基本语法
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/query-data/cte.md:11]:
```suggestion
## CTE 的基本语法
```

File: docs/nightly/zh/user-guide/query-data/cte.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+```sql
+WITH cte_name [(column1, column2, ...)] AS (
+    QUERY
+)
+SELECT ...
+FROM cte_name;
+```
+
+### 示例解释
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/query-data/cte.md:23]:
```suggestion
## 示例解释
```

File: docs/nightly/zh/user-guide/query-data/cte.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+            host, 
+            approx_percentile_cont(latency, 0.95) RANGE '5s' AS p95_latency
+          FROM 
+            grpc_latencies
+          ALIGN '5s' FILL PREV
+      ),
+      ```
+     这里我们使用[范围查询](/user-guide/query-data/sql#按时间窗口聚合数据)计算每个 `host` 在每个 5 秒时间窗口内的第 95 百分位延迟。
+     
+    - `logs`:
```
Comment (nicecui) [inline docs/nightly/zh/user-guide/query-data/cte.md:129]: The indentation here needs to be the same as `metrics`

**Comment by killme2008** [comment]
@nicecui Fixed, PTAL. Thank you. cc @fengjiachun

File: docs/nightly/en/reference/sql/with.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
+翻译成中文：# WITH
```
Comment (nicecui) [inline docs/nightly/en/reference/sql/with.md:1]:
```suggestion
# WITH
```

File: docs/nightly/en/reference/sql/with.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+Use `WITH` to specify a Common Table Expression.
+
+## What is a Common Table Expression (CTE)?
+
+A Common Table Expression (CTE) is a temporary result set that you can reference within a `SELECT`, `INSERT`, `UPDATE`, or `DELETE` statement. CTEs help to break down complex queries into more readable parts and can be referenced multiple times within the same query.
+
+## Basic Syntax of CTE
```
Comment (nicecui) [inline docs/nightly/en/reference/sql/with.md:9]:
```suggestion
## Basic syntax of CTE
```

**Review by nicecui — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**